rules/retrieve.smk

```diff
@@ -246,30 +246,42 @@ if config["enable"]["retrieve"]:
 
 
 if config["enable"]["retrieve"]:
-    current_month = datetime.now().strftime("%b")
-    current_year = datetime.now().strftime("%Y")
-    bYYYY = f"{current_month}{current_year}"
+    # Some logic to find the correct file URL
+    # Sometimes files are released delayed or ahead of schedule, check which file is currently available
 
     def check_file_exists(url):
         response = requests.head(url)
         return response.status_code == 200
 
-    url = f"https://d1gam3xoknrgr2.cloudfront.net/current/WDPA_{bYYYY}_Public.zip"
+    # Basic pattern where WDPA files can be found
+    url_pattern = (
+        "https://d1gam3xoknrgr2.cloudfront.net/current/WDPA_{bYYYY}_Public.zip"
+    )
 
-    if not check_file_exists(url):
-        prev_month = (datetime.now() - timedelta(30)).strftime("%b")
-        bYYYY = f"{prev_month}{current_year}"
-        assert check_file_exists(
-            f"https://d1gam3xoknrgr2.cloudfront.net/current/WDPA_{bYYYY}_Public.zip"
-        ), "The file does not exist."
+    # 3-letter month + 4 digit year for current/previous/next month to test
+    current_monthyear = datetime.now().strftime("%b%Y")
+    prev_monthyear = (datetime.now() - timedelta(30)).strftime("%b%Y")
+    next_monthyear = (datetime.now() + timedelta(30)).strftime("%b%Y")
+
+    # Test prioritised: current month -> previous -> next
+    for bYYYY in [current_monthyear, prev_monthyear, next_monthyear]:
```

Comment:
on the same note, if `config["enable"]["retrieve"]` is `false`, then `bYYYY` doesnt exist in `add_electricity.smk` maybe we can add the same code there ?

Reply:
Good point!

Generally, the downloaded file should drop the month in the filename. This is because otherwise, a new month would trigger the whole workflow to re-execute and invalidate all existing (intermediate) results.

I will prepare a separate pull request.